core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -398,27 +398,28 @@ namespace kagome::consensus {
 
     // create new block
     auto pre_seal_block_res =
-        proposer_->propose(best_block_hash, inherent_data, {babe_pre_digest});
+        proposer_->propose(best_block_number, inherent_data, {babe_pre_digest});
     if (!pre_seal_block_res) {
       return log_->error("Cannot propose a block: {}",
                          pre_seal_block_res.error().message());
     }
 
     auto block = pre_seal_block_res.value();
 
-    // Ensure block's extrinsics root matches extrinsics in block's body
-    BOOST_ASSERT_MSG(
-        [&block] {
-          using boost::adaptors::transformed;
-          const auto &ext_root_res = storage::trie::calculateOrderedTrieHash(
-              block.body | transformed([](const auto &ext) {
-                return common::Buffer{scale::encode(ext).value()};
-              }));
-          return ext_root_res.has_value()
-                 and (ext_root_res.value()
-                      == common::Buffer(block.header.extrinsics_root));
-        }(),
-        "Extrinsics root does not match extrinsics in the block");
+    {
+      using boost::adaptors::transformed;
+      const auto &ext_root_res = storage::trie::calculateOrderedTrieHash(
+          block.body | transformed([](const auto &ext) {
+            return common::Buffer{scale::encode(ext).value()};
+          }));
+      bool roots_match = ext_root_res.has_value()
+                         and (ext_root_res.value()
+                              == common::Buffer(block.header.extrinsics_root));
```

Comment:
Why it was extracted from macro? It should not executed in release build.

Reply:
Whoops. Debug stuff. Thanks.